designs/autoscaler.md

```diff
@@ -0,0 +1,526 @@
+
+Currently, it is impossible to have one autoscaler controller to watch multiple workload clusters with different kubeconfig at the same time, as there is no way currently to separate the pods between the workload clusters autoscaler is watching. The recommended method is to run multiple autoscalers in the management cluster with different options for how they discover the scalable resource sets.
+
+Cluster Autoscaler provides the flexibility to enable autoscaling, whether it is within the cluster, or for a remote workload cluster. Cluster Autoscaler can be:
+
+* deployed in the management cluster to autoscale the management cluster itself
+* deployed in the management cluster to autoscale a remote workload cluster
+* deployed in the workload cluster to autoscale the workload cluster itself
+
+Since EKS Anywhere follows a standard management/workload clusters setup where all the CAPI machine objects are defined in the management cluster, we choose to **deploy autoscalers in the management cluster as well — to watch and manage management cluster, and all the remote workload clusters, as default option**. In this way, we have a centralized place — management cluster, to deploy and manage all the autoscaler controllers for workload clusters.
+
+However, it is not uncommon that the management cluster size defined by the user is small, but the workload clusters are rather big, then it makes more sense to distribute the workload and run the autoscaler in each workload cluster to reduce resource usage in one management cluster. This configuration is also supported by us. Though this approach requires extra steps to create the management cluster’s kubeconfig in the workload cluster, comparing to the first approach where each workload cluster’s kubeconfig by default exists in the management cluster. Some limitations and concerns:
+
+  1. limiting user permission to the raw CAPI components: having management cluster kubeconfig in workload cluster means workload cluster user can provision and manage clusters that are not built in a conformant, reviewed or secured manner.
+  2. limiting user access to the underlying infrastructure: once non-admin workload cluster users have management cluster admin kubeconfig, which puts very high access credentials needed by CAPI. They then have high permission to the provider infrastructure (e.g vSphere: admin access to vCenter).
+  3. auditing and compliance of autoscaler: as the number of workload clusters arises, it becomes a mess of autoscaler deployments with various versions and configurations. Centralizing configuration makes sure compliance and configuration can be audited.
+  4. lack of visibility and management: it is hard to list and visualize all the autoscaler deployed in each workload cluster at the same time without user building its own monitoring tools. User needs to find credentials for each cluster and run kubectl separately to get and manage each deployment.
```

Comment:
I would maybe add something about network connectivity. One benefit of separating management and workload clusters is that you can completely cut any connection from workload to management.